Issue #19 Add style checker

MichiruKayo opened the pull request:
--added check style
--fixed check style findings.

trioletas commented on code-analysis/check-style/check-style.xml:
```diff
@@ -0,0 +1,109 @@
+<?xml version="1.0"?>
```
that's a lot of style checks..are there any presents available that we could simply reference here? just curious

MichiruKayo replied:
There are 2 presets available : 
http://checkstyle.sourceforge.net/style_configs.html Google and Sun, will check how to include them by default

MichiruKayo replied:
unfortunately it is not possible to have 2 configs . Issue is open in style checker :https://github.com/checkstyle/checkstyle/issues/991

trioletas commented on api/src/main/java/lv/ctco/tpl/bff/integration/icndb/jokes/JokeValueModel.java:
```diff
@@ -6,7 +6,7 @@
-    String id;
-    String joke;
-    List<String> categories;
+    private String id;
```
can't say I like this one: _who cares_ if it's not private? It's just a stupid java bean

trioletas commented on api/build.gradle:
```diff
@@ -24,10 +24,30 @@ apply plugin: 'org.springframework.boot'
+  def dir = "${parent.projectDir}/code-analysis/check-style"
+  toolVersion = '8.2'
+  configProperties.configDir = dir
+  configFile = file("$dir/check-style.xml")
```
Can we use the default file location instead and just drop this line? I would like to reduce boilerplate configs as much as possible, let's stick with convention over configuration as much as possible please

trioletas commented on api/build.gradle:
```diff
@@ -24,10 +24,30 @@ apply plugin: 'org.springframework.boot'
 
+apply plugin: 'checkstyle'
+apply plugin: 'pmd'
+pmdTest.enabled = false
```
why?

MichiruKayo replied:
In tests we usually have hard coded strings/ magic numbers, and style check fails in such case..

trioletas replied:
can we use inline directives to suppress the linting for such cases? tests are also code so we should treat them as such

trioletas commented on README.md:
```diff
@@ -29,6 +29,18 @@ Spring Boot, GraphQL template project with batteries included.
+
+#### IntelliJ
+
+- Edit configurations > "Application" > Add task in "Before launch" section "checkstyleMain pmdMain" and place it as first
```
is necessary ? can we just use the plugin and be done with that?

MichiruKayo replied:
If you start using intellij idea Application run/debug then plugin is not executed.. couldn't find how to force it..

trioletas replied:
Okay let's think about that. But please remove these instructions from the documentation. its unnecessarily cumbersome

trioletas commented on README.md:
```diff
@@ -29,6 +29,18 @@ Spring Boot, GraphQL template project with batteries included.
+
+- Edit configurations > "Application" > Add task in "Before launch" section "checkstyleMain pmdMain" and place it as first
+- Additionally you can download "CheckStyle plugin". Then open Settings > Other Settings > Checkstyle. Add path to check-style.xml file 
+  and provide link to check-style folder in "${configDir}". This will display errors in editor, but it wont affect build/run. 
```
> but it won affect build/run
sounds fine by me :)

trioletas commented on README.md:
```diff
@@ -29,6 +29,18 @@ Spring Boot, GraphQL template project with batteries included.
+
+#### Gradle
+
+- As default behaviour it is configured to call "checkstyleMain pmdMain" before "bootRun" task is executed
```
Not sure I like this one, it basically stops me from launching the dev mode unless all my modifiers are private and stuff.. Can we relax this one?

trioletas commented on api/build.gradle:
```diff
@@ -24,10 +24,30 @@ apply plugin: 'org.springframework.boot'
+  toolVersion = '8.2'
+  configProperties.configDir = dir
+  configFile = file("$dir/check-style.xml")
+  sourceSets = [] // used to skip tests
```
why?

trioletas commented on api/build.gradle:
```diff
@@ -24,10 +24,30 @@ apply plugin: 'org.springframework.boot'
+pmd {
+  toolVersion = '5.8.1'
+  consoleOutput = true
+  ruleSetFiles = files("${parent.projectDir}/code-analysis/pmd/pmd.xml")
```
I think we can do a convention over configuration with this one as well. I do recall there being a default directory where pmd looks for its configs

trioletas commented on code-analysis/check-style/check-style.xml:
```diff
@@ -0,0 +1,109 @@
+    </module>
+
+    <module name="Header">
+        <property name="header" value="Copyright Swiss Reinsurance Company, Mythenquai 50/60, CH 8022 Zurich. All rights reserved."/>
```
oops you did it again

please be careful with this stuff

trioletas commented on code-analysis/check-style/suppressions.xml:
```diff
@@ -0,0 +1,9 @@
+    "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+
+<suppressions>
+    <!--Add exclusion here (e.g <suppress files="JokeResolver.java" checks="MethodLength"/>) -->
```
Are we trying to teach how to use `check-style` here? Please drop all unnecessary config files, thanks!

MichiruKayo commented:
1: Removed pmd checks (not worth it)
2: Based on rij checks, reviewed them and removed ones i think are not needed
3: updated read.me
4:  config location is required. Default path is <build.gradle.folder>/config/stylecheck/stylecheck.xml, in our case it is api/config/... , dont think adding such folder is good.. maybe better to change it to root build.gradle? ( but i do not know how to configure that)
5: Changed dependendOn to "build" will probably be called during deployment?.. not sure where to attach it, since we do not have many tasks..

trioletas commented:
>4: config location is required. Default path is <build.gradle.folder>/config/stylecheck/stylecheck.xml, > in our case it is api/config/... , dont think adding such folder is good.. maybe better to change it to > > root build.gradle? ( but i do not know how to configure that)
I do not see a problem with the `api/config` folder. It is a configuration for the api module is it not?

MichiruKayo commented:
Changed to default location as we discussed offline

trioletas commented:
Thanks, looks good to me.
@sergeytrasko any comments from your side? Otherwise we're good to go.
